github/utils.go

```diff
@@ -40,15 +55,61 @@ func connect(ctx context.Context, d *plugin.QueryData) *github.Client {
 		baseURL = *githubConfig.BaseURL
 	}
 
-	if token == "" {
-		panic("'token' must be set in the connection configuration. Edit your connection configuration file and then restart Steampipe")
+	//// Github App authentication.
+	if githubConfig.AppId != nil {
+		githubAppId = *githubConfig.AppId
+	}
+	if githubConfig.InstallationId != nil {
+		githubInstallationId = *githubConfig.InstallationId
+	}
+	if githubConfig.PrivateKey != nil {
+		privateKeyPath = *githubConfig.PrivateKey
+	}
+
+	if token == "" && (githubAppId == "" || githubInstallationId == "" || privateKeyPath == "") {
+		panic("'token' or 'app_id', 'installation_id' and 'private_key' or `app_installation_access_token` must be set in the connection configuration. Edit your connection configuration file and then restart Steampipe")
+	}
+
+	// Return error for unsupported token by prefix
+	if token != "" && !strings.HasPrefix(token, "ghs_") && !strings.HasPrefix(token, "ghp_") {
+		panic("Supported token formats are 'ghp_' and 'ghp_'")
 	}
 
-	ts := oauth2.StaticTokenSource(
-		&oauth2.Token{AccessToken: token},
-	)
-	tc := oauth2.NewClient(ctx, ts)
-	conn := github.NewClient(tc)
+	var client *github.Client
+
+	// Authentication with Github access token
+	if token != "" && strings.HasPrefix(token, "ghp_") {
+		ts := oauth2.StaticTokenSource(
+			&oauth2.Token{AccessToken: token},
+		)
+		tc := oauth2.NewClient(ctx, ts)
+		client = github.NewClient(tc)
+	}
+
+	// Authentication Using App Installation Access Token
+	if token != "" && strings.HasPrefix(token, "ghs_") {
+		client = github.NewClient(&http.Client{Transport: &oauth2Transport{
+			Token: token,
+		}})
+	}
+
+	// Authentication as Github APP Installation authentication
+	if githubAppId != "" && githubInstallationId != "" && privateKeyPath != "" && token == "" {
+		ghAppId, err := strconv.ParseInt(githubAppId, 10, 64)
+		if err != nil {
+			panic(err)
+		}
+		ghInstallationId, err := strconv.ParseInt(githubInstallationId, 10, 64)
+		if err != nil {
+			panic(err)
+		}
+		itr, err := ghinstallation.NewKeyFromFile(http.DefaultTransport, ghAppId, ghInstallationId, privateKeyPath)
```

Comment:
Should we add in a function to check if privateKeypath exists or not? Or is that taken care of by `ghinstallation.NewKeyFromFile` function?